script/setup.js

```diff
@@ -37,37 +36,48 @@ try {
 		strict: false,
 	});
 
-	async function getPrefillOrPromptedValue(key, message) {
+	async function getPrefillOrPromptedValue(key, message, placeholder) {
 		const { [key]: value } = values[key]
 			? (console.log(chalk.grey(`Pre-filling ${key} to ${values[key]}.`)),
 			  values)
-			: await prompt({
+			: await text({
 					message,
-					name: key,
-					type: "input",
+					placeholder,
+					validate: (val) => {
+						if (val.length === 0) return "Please enter a value.";
+					},
 			  });
 
+		if (isCancel(value)) {
+			cancel("Operation cancelled. Exiting setup - maybe another time? 👋");
+			process.exit(0);
+		}
+
 		return value;
 	}
 
 	const repository = await getPrefillOrPromptedValue(
 		"repository",
-		"What will the kebab-case name of the repository be?"
+		"What will the kebab-case name of the repository be?",
+		"my-repo"
```

Comment:
Adding some flavor here 😄 

```suggestion
		"my-lovely-repository"
```